draft-ietf-httpbis-messaging-latest.xml

```diff
@@ -1739,6 +1739,11 @@ https://www.example.org
    ordered request that has not yet received a final (non-<x:ref>1xx</x:ref>)
    response.
 </t>
+<t>
+   If a HTTP/1.1 client receives data on a connection that doesn't have any
+   outstanding requests, it &MUST-NOT; consider them to be a response to a
+   not-yet-issued request; it &SHOULD; silently discard them.
```

Comment:
...and SHOULD perhaps close the connection, since it might now be out-of-sync with the server about where the message boundaries are.

Reply:
Is it really necessary to say it SHOULD silently discard them? What it should do in this case is dependent on the nature of the client, not on the protocol.

Yes, the connection MUST be closed. OTOH, do implementations ever check to see if data has arrived before they send a first request? My guess is this would depend on what connection management and IO frameworks are being used to do connection management. Let's try to avoid making simple implementations impossible.

Reply:
how does the should interact with 408? 408 seems like a pretty common pattern for this

Reply:
I removed the "silent discard." Regarding 408s -- you might encounter one, but the server is going to be closing the connection anyway, right?